src/main/java/eu/greev/dcbot/ticketsystem/service/Transcript.java

```diff
@@ -1,135 +1,114 @@
-                        } catch (ArrayIndexOutOfBoundsException e) { return; }
-                        writer.write(log);
-                        writer.newLine();
-                    } else {
-                        writer.write(line);
-                        writer.newLine();
+                builder.append("[").append(message.getAuthor()).append("] ").append(message.getOriginalContent());
+                List<Edit> edits = message.getEdits();
+                if (!edits.isEmpty()) {
+                    builder.append(" | Edits:");
+
+                    for (int i = 0; i < edits.size(); i++) {
+                        builder.append(" ").append(edits.get(i).getEdit());
+
+                        if (edits.size() - 1 != i) {
+                            builder.append(" ->");
+                        }
                     }
-                    transcript.delete();
-                    temp.renameTo(new File("./Tickets/transcripts/" + id + ".txt"));
                 }
-            }
-        } catch (IOException e) {
-            log.error("Could not read transcript of ticket #" + id, e);
-        }
-    }
 
-    public File clean() {
-        File temp = new File("./Tickets/transcripts/" + id + ".temp");
-        try {
-            temp.createNewFile();
-            try (BufferedWriter writer = new BufferedWriter(new FileWriter(temp, true));
-                 BufferedReader reader = new BufferedReader(new FileReader(transcript));) {
-                List<String> lines = reader.lines().toList();
-
-                for (String line : lines) {
-                    if (lines.get(1).equals(line)) continue;
-                    if (lines.get(0).equals(line)) {
-                        writer.write("Transcript of ticket: #" + id);
-                        writer.newLine();
-                        continue;
-                    }
-                    if (!line.contains("} ")) {
-                        writer.write(line);
-                        writer.newLine();
-                        continue;
-                    }
-                    String content = line.split("} ")[1];
-                    writer.write(content);
-                    writer.newLine();
+                if (message.isDeleted() && message.getId() != 0) {
+                    builder.insert(0, "~~").append("~~");
                 }
+                writer.write(builder.toString());
+                writer.newLine();
             }
-            transcript.delete();
-            temp.renameTo(transcript);
         } catch (IOException e) {
-            log.error("Could not clean transcript of ticket #" + id, e);
+            log.error("Could not clean transcript of ticket #" + ticketId, e);
         }
         return transcript;
     }
+
+    private String formatTimestamp(long timestamp) {
+        return new SimpleDateFormat("[hh:mm:ss a '|' dd'th' MMM yyyy] ").format(new Date(timestamp * 1000));
```

Comment:
yyyy.MM.dd please

Reply:
Also I think there should be a better Instant method for converting a timestamp to a date